addons/point_of_sale/models/pos_order.py

```diff
@@ -470,14 +470,18 @@ def action_pos_order_paid(self):
     def _prepare_invoice_vals(self):
         self.ensure_one()
         timezone = pytz.timezone(self._context.get('tz') or self.env.user.tz or 'UTC')
+        narration = self.note
+        if not narration and self.env['ir.config_parameter'].sudo().get_param('account.use_invoice_terms'):
+            narration_context = self.env['account.move'].with_context(lang=self.partner_id.lang or self.env.lang)
+            narration = narration_context._get_company_terms(self.company_id) or narration_context._get_company_terms(self.env.company)
```

Comment:
Only provide the value in the dict if `self.note` is not Falsy instead of copŷ/pasting the compute method here, that way it will be recomputed automatically